modules/pubsub/src/main/resources/application.conf

```diff
@@ -4,15 +4,21 @@
     "parallelPullCount": 1
     "maxQueueSize": 3000
     "maxRequestBytes": 50000000
-    "maxAckExtensionPeriod": 1 hour
+    "maxAckExtensionPeriod": 1 hour,
+    "gcpUserAgent": {
+      "productName": "Snowplow OSS"
```

Comment:
> I find it unsatisfying that we need to set the productName in four different places. Which means we will need to override it in four different places.

Yeah, I agree with this sentiment. I will give it thought 👍 

> Did you consider calling the config field userAgent instead of gcpUserAgent? I know you called it gcpUserAgent in the collector, but that's because in the collector it was in a non-gcp-specific part of the config.

I would prefer to leave it as `gcpUserAgent` in here as well because it signals that user-agent is gcp related instead of being snowplow related.